src/data/set/function.lean

```diff
@@ -666,6 +666,22 @@ lemma piecewise_eq_on (f g : α → β) : eq_on (s.piecewise f g) f s :=
 lemma piecewise_eq_on_compl (f g : α → β) : eq_on (s.piecewise f g) g sᶜ :=
 λ _, piecewise_eq_of_not_mem _ _ _
 
+lemma piecewise_le {δ : α → Type*} [Π i, preorder (δ i)] {s : set α} [Π j, decidable (j ∈ s)]
```

Comment:
Which spelling is more canonical? Yours or this one?
```suggestion
lemma piecewise_le {δ : α → Type*} [Π i, preorder (δ i)] {s : set α} [decidable_pred (∈ s)]
```

Reply:
`git grep` finds both, my spelling gives more matches. In particular, the definition of `set.piecewise` uses my spelling, so I think that we should keep it in this PR.